R/Data-class.R

```diff
@@ -141,58 +141,42 @@ Data <- function(x = numeric(),
   )
 }
 
-## --------------------------------------------------
-## Subclass with additional biomarker information
-## --------------------------------------------------
+# DataDual-class ----
 
-##' Class for the dual endpoint data input
-##'
-##' This is a subclass of \code{\linkS4class{Data}}, so contains all
-##' slots from \code{\linkS4class{Data}}, and in addition biomarker
-##' values.
-##'
-##' @slot w the continuous vector of biomarker values
-##'
-##' @example examples/Data-class-DataDual.R
-##' @export
-##' @keywords classes
-.DataDual <-
-    setClass(Class="DataDual",
-             representation=
-                 representation(w="numeric"),
-             contains="Data",
-             validity=
-                 function(object){
-                     o <- Validate()
+#' `DataDual`
+#'
+#' @description `r lifecycle::badge("stable")`
+#'
+#' [`DataDual`] is a class for the dual endpoint data.
+#' It inherits from [`Data`] and it contains additional biomarker information.
+#'
+#' @slot w (`numeric`)\cr the continuous vector of biomarker values.
+#'
+#' @aliases DataDual
+#' @export
+.DataDual <- setClass(
+  Class = "DataDual",
+  slots = c(w = "numeric"),
+  contains = "Data",
+  validity = validate_data_dual
+)
 
-                     o$check(identical(object@nObs, length(object@w)),
-                             "w must have length nObs")
+# DataDual-constructor ----
 
-                     o$result()
-                 })
-validObject(.DataDual())
+#' @rdname DataDual-class
+#'
+#' @param w (`numeric`)\cr the continuous vector of biomarker values.
+#' @param ... parameters passed to [`Data`].
```

Comment:
```suggestion
#' @param ... parameters passed to [Data()].
```
(to the constructor)